samples/besu/simpleasset/app/AssetExchangeERC20.js

```diff
@@ -0,0 +1,308 @@
+    )
+    await assertTokenBal(
+      BobERC20,
+      Bob,
+      senderInitialBalance,
+      'balance not transferred to Bob in before()'
+    )
+    await assertTokenBal(
+      BobERC20,
+      Alice,
+      0,
+      'Alice should not have any BobERC20 tokens in before()'
+    )
+*/
+}
+
+
+// Withdrawal function of a fungible asset/token of type tokenContractId 
+// and amount tokenAmount locked using contractId for a receiver. 
+// hashSecret is the secret to unlock the asset corresponding to the commitment 
+// used when locking it
+async function withdraw(htlc, contractId, tokenContractId, tokenAmount, receiver, hashSecret) {
+    await htlc.claimFungibleAsset(contractId, hashSecret, {
+      from: receiver,
+    })
+
+/*
+    // Check tokens now owned by Alice
+    await assertTokenBal(
+      tokenContractId,
+      receiver,
+      tokenAmount,
+      `${receiver} doesn't not own ${tokenAmount} tokens`
+    )
+*/
+
+    const contractArr = await htlc.getContract.call(contractId)
+    const contract = htlcERC20ArrayToObj(contractArr)
+    assert.isTrue(contract.withdrawn) // withdrawn set
+    assert.isFalse(contract.refunded) // refunded still false
+    assert.equal(contract.preimage, hashSecret)
+
+	return contract.withdrawn // the withdrawn flag is set iff the withdraw function is successful
+}
+
+
+const assertTokenBal = async (token, addr, tokenAmount, msg) => {
+    assertEqualBN(
+      await token.balanceOf.call(addr),
+      tokenAmount,
+      msg ? msg : 'wrong token balance'
+    )
+}
+
+// Refund function
+async function refund(htlc, contractId, tokenContract, sender, initialBalance, tokenAmount) {
+    const currentBalance = initialBalance - tokenAmount
+
+      let result = await htlc.unlockFungibleAsset(contractId, {
+        from: sender
+      })
+
+      // verify the event was emitted
+      //truffleAssert.eventEmitted(result, 'HTLCERC20Refund', ev => {
+      //  return ev.contractId === a2bSwapId
+      //}, `Refunded Alice`)
+
+      //await assertTokenBal(tokenContract, sender, currentBalance)
+
+}
+
+
+// Function to create a new smart contract for locking asset for HTLC
+// 'intitiator' locks 'tokenAmount' number of tokens of type 'token' for
+// 'counterparty' using the contract constructs of 'htlc'. 
+// 'config' contains the locking guidelines and parameters
+const newHtlc = async (token, tokenAmount, htlc, config, initiator, counterparty) => {
+    // initiator of the swap has to first designate the swap contract as a spender of his/her money
+    // with allowance matching the swap amount
+    await token.approve(htlc.address, tokenAmount, {from: initiator})
+    return htlc.lockFungibleAsset(
+      counterparty,
+      config.hashlock,
+      config.timelock,
+      token.address,
+      tokenAmount,
+      {
+        from: initiator,
+      }
+    )
+}
+  
+/*
+// Sender has the original secret, calls withdraw with the secret to claim the EU tokens
```

Comment:
Why is this commented out? There's a `withdraw` function further above. If this is not needed anymore, perhaps delete it.